post-processor/vagrant/post-processor.go

```diff
@@ -190,12 +191,20 @@ func (p *PostProcessor) PostProcessProvider(name string, provider Provider, ui p
 	var customVagrantfile string
 	if config.VagrantfileTemplate != "" {
 		ui.Message(fmt.Sprintf("Using custom Vagrantfile: %s", config.VagrantfileTemplate))
-		customBytes, err := ioutil.ReadFile(config.VagrantfileTemplate)
-		if err != nil {
-			return nil, false, err
+
+		var templateContent string
```

Comment:
I believe this condition might be nested in a conflicting if statement. 

```go
// lets default to using VagrantTemplateContent
templateContent := config.VagrantfileTemplateContent
if templateContent == "" && config.VagrantfileTemplate != "" {
	ui.Message(fmt.Sprintf("Using custom Vagrantfile: %s", config.VagrantfileTemplate))
	customBytes, err := ioutil.ReadFile(config.VagrantfileTemplate)
	if err != nil {
		return nil, false, err
	}
	templateContent = string(customBytes)
}
	
if templateContent != "" {
       customVagrantfile, err = interpolate.Render(templateContent, &config.ctx)
	if err != nil {
		return nil, false, err
	}
}

```

Reply:
@nywilken  could you please see the below comment https://github.com/hashicorp/packer-plugin-vagrant/pull/112#discussion_r1573626603 and let me know if this is still needs to be changed !!